Source/Core/DolphinQt2/GameList/GameList.cpp

```diff
@@ -474,6 +603,52 @@ std::shared_ptr<const UICommon::GameFile> GameList::GetSelectedGame() const
   return {};
 }
 
+QList<std::shared_ptr<const UICommon::GameFile>> GameList::GetSelectedGames() const
+{
+  QAbstractItemView* view;
+  QSortFilterProxyModel* proxy;
+  if (currentWidget() == m_list)
+  {
+    view = m_list;
+    proxy = m_list_proxy;
+  }
+  else
+  {
+    view = m_grid;
+    proxy = m_grid_proxy;
+  }
+  QList<std::shared_ptr<const UICommon::GameFile>> selected_list;
+  QItemSelectionModel* sel_model = view->selectionModel();
+  if (sel_model->hasSelection())
+  {
+    QModelIndexList index_list =
+        currentWidget() == m_list ? sel_model->selectedRows() : sel_model->selectedIndexes();
+    for (int i = 0; i < index_list.size(); i++)
+    {
+      QModelIndex model_index = proxy->mapToSource(index_list[i]);
+      selected_list.push_back(m_model->GetGameFile(model_index.row()));
+    }
+  }
+  return selected_list;
+}
+
+bool GameList::HasMultipleSelected() const
+{
+  if (currentWidget() == m_list)
+  {
+    QItemSelectionModel* sel_model = m_list->selectionModel();
+    if (sel_model->hasSelection())
+      return sel_model->selectedRows().size() > 1;
+  }
+  else
+  {
+    QItemSelectionModel* sel_model = m_grid->selectionModel();
+    if (sel_model->hasSelection())
+      return sel_model->selectedIndexes().size() > 1;
```

Comment:
This `if` is the same in both branches, any thoughts on compacting this a little to avoid the duplication?

This could probably work (not sure if you'd need a cast on `currentWidget()` or something else there though):
```c++
QItemSelectionModel* sel_model = currentWidget()->selectionModel();
if (sel_model->hasSelection())
  return sel_model->selectedRows().size() > 1;
```

Reply:
Haha yeah I realize now that's a bit silly, I'll fix that.

Reply:
Turns out it does require a cast because that function returns a generic QWidget.

Reply:
No common base class for those two? You can still do a ternary then:
```c++
QItemSelectionModel* sel_model = currentWidget() == m_list ? m_list->selectionModel() : m_grid->selectionModel();
```

Reply:
Ah, but wait, inside the `if` block is a different function call. `selectedIndexes` and `selectedRows`, because the behavior is different between the table and grid views.

Reply:
I think the `hasSelection` check isn't needed in this case as it should just return a list of size 0, and can probably be reduced entirely to a simple ternary.

Reply:
Eep, my brain triggered a pattern match, but kinda missed the fact that it's not the same calls. Ignore this one then ;)